tox.ini

```diff
@@ -1,10 +1,16 @@
 [tox]
-envlist = lint,py27,py35,py36,py37,py38,py39,docs
+envlist = lint,docs,py27-ipython5,py35-ipython{5,6,7},py36-ipython{5,6,7},py37-ipython{6,7},py3{8,9,10}-ipython{7,8}
 
 [testenv]
+deps:
+  ipython5: IPython>=5,<6
+  ipython6: IPython>=6,<7
+  ipython7: IPython>=7,<8
+  ipython8: IPython>=8
+
 extras = tests
 commands = pytest -s {posargs}
-passenv = HOME SHELL CI
+passenv = HOME SHELL CI DOITLIVE_INTERPRETER
```

Comment:
question (non-blocking): is this change necessary?

Reply:
Yeah. In my setup, the default shell is `fish`, but the tested scripts use bash. Since fish is not bash (or POSIX) compatible, this breaks, so when testing locally,  I wanted to override `fish` with `bash`.